src/main/java/seedu/storage/Storage.java

```diff
@@ -0,0 +1,42 @@
+package seedu.storage;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import seedu.equipment.DuplicateSerialNumber;
+import seedu.equipment.Equipment;
+import seedu.equipment.EquipmentManager;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class Storage {
+    public final String path = "./equipments.json";
+    private final Gson gson = new Gson();
+
+    public void loadData(EquipmentManager equipmentManager) throws DuplicateSerialNumber {
+        try {
+            Reader reader = Files.newBufferedReader(Path.of(path));
+            List<Equipment> equipments = gson.fromJson(reader, new TypeToken<List<Equipment>>() {}.getType());
+            for (Equipment equipment : equipments) {
+                equipmentManager.addEquipment(equipment);
+            }
+        } catch (IOException e) {
+            System.out.println("io1");
```

Comment:
Do we know what exactly can throw the IOException? Like is it from `gson.fromJson`? If so the IOException can be more descriptive of the error and suggest how to resolve it.